src/jitlayers.cpp

```diff
@@ -196,23 +194,14 @@ static jl_callptr_t _jl_compile_codeinst(
         start_time = jl_hrtime();
 
     assert(jl_is_code_instance(codeinst));
-#ifndef NDEBUG
-    size_t max_world = jl_atomic_load_relaxed(&codeinst->max_world);
-    assert(jl_atomic_load_relaxed(&codeinst->min_world) <= world && (max_world >= world || max_world == 0) &&
-        "invalid world for method-instance");
-#endif
 
     JL_TIMING(CODEINST_COMPILE, CODEINST_COMPILE);
-#ifdef USE_TRACY
-    if (is_recompile) {
-        TracyCZoneColor(JL_TIMING_DEFAULT_BLOCK->tracy_ctx, 0xFFA500);
-    }
-#endif
     jl_callptr_t fptr = NULL;
     // emit the code in LLVM IR form
     jl_codegen_params_t params(std::move(context), jl_ExecutionEngine->getDataLayout(), jl_ExecutionEngine->getTargetTriple()); // Locks the context
     params.cache = true;
-    params.world = world;
+    params.min_world = jl_atomic_load_relaxed(&codeinst->min_world);
+    params.max_world = jl_atomic_load_relaxed(&codeinst->max_world);
```

Comment:
:+1: to this bug being fixed finally. I was assuming in the past we would go the other direction to fix this issue (e.g. narrowing the `codeinst->min_world`/`codeinst->max_world` applicability based on what became possible to compile). But it seems this direction should work too.